src/openforms/prefill/__init__.py

```diff
@@ -230,27 +226,27 @@ def _set_default_values(
     is inspected for prefill configuration, which is then looked up in
     ``prefilled_values`` to set the component ``defaultValue``.
     """
-    if "prefill" in configuration:
-        default_value = configuration.get("defaultValue")
-        prefill_value = prefilled_values.get(
-            configuration["prefill"]["plugin"], {}
-        ).get(configuration["prefill"]["attribute"])
+    for component in iter_components(configuration, recursive=True):
+        if not (prefill := component.get("prefill")):
+            continue
+        if not (plugin := prefill.get("plugin")):
+            continue
+        if not (attribute := prefill.get("attribute")):
+            continue
 
+        default_value = configuration.get("defaultValue")
```

Comment:
I think this should be
```suggestion
        default_value = component.get("defaultValue")
```